### PR TITLE
tracing: instrument outcome proof computation

### PR DESCRIPTION
**Pull request by bowenwang1996**
Adding instrumentation on outcome proof computation, which may take a non-negligible amount of time if there are thousands of outcomes.

### CODE REVIEW

**Comment by bowenwang1996** [comment]
@race-of-sloth

**Comment by race-of-sloths** [comment]
@bowenwang1996 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/bowenwang1996?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/bowenwang1996?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/bowenwang1996)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/bowenwang1996)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by wacban — APPROVED** [review]
LGTM

**Review by tayfunelmas — APPROVED** [review]

**Comment by race-of-sloths** [comment]
### 🔄 The PR has been merged.
>[!IMPORTANT]
This pull request is a part of the Race of Sloths and has not been scored yet. Scoring will close in 24 hours! 🕰️

**Comment by frol** [comment]
@wacban @tayfunelmas Please, score this PR with `@race-of-sloths score [0, 1, 2, 3, 5, 8, 13]` command (see [scoring criteria](https://race-of-sloths.com/faq))

**Comment by race-of-sloths** [comment]
### 🏆 Score recorded! 
It looks like @frol provided a score of [0, 1, 2, 3, 5, 8, 13]` command (see [scoring criteria](https://race-of-sloths.com/faq)), but based on our Fibonacci scoring rules, we've adjusted it to 0.
> [!TIP]
We accept only the following Fibonacci numbers [0, 1, 2, 3, 5, 8, 13] for scoring

**Comment by wacban** [comment]
@race-of-sloths score 1

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@wacban, thank you for scoring this pull request in the Race of Sloths!

**Comment by frol** [comment]
@race-of-sloths score 2

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 1!
You have received 20 (10 base + 10 weekly bonus) Sloth points for this contribution


Another weekly streak completed, well done @bowenwang1996! To keep your weekly streak and get another bonus make pull request next week! Looking forward to see you in race-of-sloths

**Pull request merged**
